staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -1033,6 +1033,12 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericapifilters.WithImpersonation(handler, c.Authorization.Authorizer, c.Serializer)
 	handler = filterlatency.TrackStarted(handler, c.TracerProvider, "impersonation")
 
+	if c.FeatureGate.Enabled(genericfeatures.ConstrainedImpersonation) {
+		handler = filterlatency.TrackCompleted(handler)
+		handler = genericapifilters.WithContrainedImpersonation(handler, c.Authorization.Authorizer, c.Serializer)
+		handler = filterlatency.TrackStarted(handler, c.TracerProvider, "constrainedimpersonation")
+	}
```

Comment:
These two should be mutually exclusive:

```go
	if c.FeatureGate.Enabled(genericfeatures.ConstrainedImpersonation) {
		handler = filterlatency.TrackCompleted(handler)
		handler = genericapifilters.WithContrainedImpersonation(handler, c.Authorization.Authorizer, c.Serializer)
		handler = filterlatency.TrackStarted(handler, c.TracerProvider, "constrainedimpersonation")
	} else {
		handler = filterlatency.TrackCompleted(handler)
		handler = genericapifilters.WithImpersonation(handler, c.Authorization.Authorizer, c.Serializer)
		handler = filterlatency.TrackStarted(handler, c.TracerProvider, "impersonation")
	}
```

`ConstrainedImpersonation` is expected to handle all impersonation logic, including the legacy flow.